garden_ai/gardens.py

```diff
@@ -142,6 +142,24 @@ def pipeline_names(self) -> List[str]:
             names += [name]
         return names
 
+    def _set_pipelines_from_remote_metadata(self, pipeline_metadata: list[dict]):
+        """
+        Given a list of dicts in RegisteredPipeline format (as from Globus Search),
+        attempt to convert them to RegisteredPipelines and use them to populate _pipelines.
+        """
+        pipelines = []
+        for meta in pipeline_metadata:
+            try:
+                pipeline = RegisteredPipeline(**meta)
+                pipeline._env_vars = self._env_vars
+                pipelines.append(pipeline)
+            except ValidationError as e:
+                logger.warning(
+                    f"Could not parse pipeline: {json.dumps(meta)}. {e.__str__()}"
+                )
```

Comment:
if you add a bare `raise` here it'll also re-raise the validation error, which is usually pretty-printed/tries to be human readable